bukkit-legacy/src/main/java/com/viaversion/viaversion/bukkit/util/LegacyBlockToItem.java

```diff
@@ -27,20 +27,15 @@
 import java.util.logging.Level;
 
 public class LegacyBlockToItem {
-    private static final LegacyBlockToItem INSTANCE;
+
     private static final Random RANDOM = new Random();
+    private static LegacyBlockToItem INSTANCE;
 
     static {
-        LegacyBlockToItem instance = null;
         try {
-            instance = new LegacyBlockToItem();
-        } catch (ReflectiveOperationException ex) {
-            Via.getPlatform().getLogger().log(
-                Level.WARNING,
-                "Couldn't find reflection methods/fields for pick-block functionality.\n" +
-                    "Pick-block won't work on clients 1.21.4 and newer.", ex);
+            INSTANCE = new LegacyBlockToItem();
+        } catch (ReflectiveOperationException ignored) {
         }
```

Comment:
This should not silently fail if a server removes or changes any of the reflection used here, you should keep the previous code structure and only replace the isItem method check instead. like your previous suggestion for a simple version check

Reply:
The reason why i decided to use a different approach is that the current implementation only supports < 1.8 and will very likely never be supported on any version ever again. So this *should* be a very safe way for checking.
But i still agree that it is kinda dirty. I will implement a version check soon.